client/ayon_core/tools/sceneinventory/model.py

```diff
@@ -150,7 +153,7 @@ def refresh(self, selected=None):
             if repre_info.is_valid
         }
         version_items_by_product_id = self._controller.get_version_items(
-            product_ids
+            product_ids, project_names
         )
```

Comment:
I think this is the wrong approach because we'd need to know which product id belongs to which project. What @iLLiCiTiT mentioined before is to separate the query to `self._controller.get_version_items()` to have separate calls per project. So above, we'd get the product ids per project and then here we would do:

```python
version_items_by_product_id = self._controller.get_version_items(project_name, product_id)
```

As stated before - as far as I know on the backend nothing is keeping the database from having a product entity with the same `id` across different projects. Is that correct @martastain ? (It's unlikely by just creating products to happen, but when cloning e.g. projects I suppose it may very well share the same ids?) As such, we should still make sure to be explicit about which project each entry belongs to.

The same goes for `version_items_by_product_id` dict. That should technically also be "per project" because otherwise `product_id` may overlap accidentally.

Reply:
Yes